packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -239,11 +239,35 @@ export class InternalModelManager {
   constructor(
     private readonly apiIdentifier: string,
     readonly connection: GadgetConnection,
-    readonly options?: { pluralApiIdentifier: string }
+    readonly options?: { pluralApiIdentifier: string; hasAmbiguousIdentifiers: boolean }
   ) {
     this.capitalizedApiIdentifier = camelize(apiIdentifier);
   }
 
+  private validateRecord(record: RecordData) {
+    if (!this.options || !this.options.hasAmbiguousIdentifiers) {
+      return true;
+    }
+    const keys = Object.keys(record);
+
+    // any keys that aren't the apiIdentifier are invalid
+    return !keys.some((key) => key !== this.apiIdentifier);
```

Comment:
nit: I find it takes me a bit to reason about multiple negations ("valid if not some key is not the api identifier"). What about:

```js
return keys.every((key) => key === this.apiIdentifier);
```